Validation/Geometry/src/MaterialBudgetAction.cc

```diff
@@ -317,18 +315,13 @@ void MaterialBudgetAction::update(const EndOfTrack* trk)
 //-------------------------------------------------------------------------
 void MaterialBudgetAction::update(const EndOfRun* )
 {
-  if (saveToTxt) delete theTxt;
-  if (saveToTree) delete theTree;
-  if (saveToHistos) delete theHistos;
-  if (theHistoMgr) delete theHistoMgr;
-  delete theData;
```

Comment:
Hello @ianna 

The (required) deletion of this `delete` statements does not allow to create theHistos ROOT File some of the configuration files, the `save` method is (wrongly?) located in the destructor of `MaterialBudgetTrackerHistos` (not in this commit though) as you can see here:

https://github.com/cms-sw/cmssw/blob/fb7c415b2b63673b87d631ff2a16966d1c428180/Validation/Geometry/interface/MaterialBudgetTrackerHistos.h#L14

Which runs:

https://github.com/cms-sw/cmssw/blob/fb7c415b2b63673b87d631ff2a16966d1c428180/Validation/Geometry/src/MaterialBudgetTrackerHistos.cc#L739

As it is right now no ROOT file is being created by `runP_Tracker_cfg.py` (and probably not by `runP_HGCal_cfg.py` either @apsallid ). 

Please check this proposed solution:

 [`saveToFile` method](https://github.com/vargasa/cmssw/commit/2600a9b78b38dd0f880c6b3fac5e6eedade037d1) in `MaterialBudgetFormat` 

But will still need to see which other parts of the code should be modified. If you have any suggestion or comments please let me know

Thanks!